java/clients/src/main/java/sleeper/clients/deploy/properties/GeneratePropertiesDocumentation.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2022-2025 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.clients.deploy.properties;
+
+import sleeper.clients.util.table.TableWriter;
+import sleeper.core.properties.PropertyGroup;
+import sleeper.core.properties.SleeperProperties;
+import sleeper.core.properties.instance.InstanceProperties;
+import sleeper.core.properties.instance.InstancePropertyGroup;
+import sleeper.core.properties.table.TableProperties;
+import sleeper.core.properties.table.TablePropertyGroup;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Locale;
+import java.util.function.Consumer;
+
+public class GeneratePropertiesDocumentation {
+
+    private GeneratePropertiesDocumentation() {
+    }
+
+    public static void generateDocumentation(Path root) throws IOException {
+        generateMasterPage(root);
+        Path headPath = Files.createDirectories(root.resolve("docs/usage/properties"));
+
+        //---------- Instance Properties ----------
+        Path instancePath = Files.createDirectories(headPath.resolve("instance/"));
+        InstancePropertyGroup.getAll().forEach(instancePropertyGroup -> {
+            try {
+                writeFile(instancePath.resolve(groupNameToFileName(instancePropertyGroup.getName())),
+                        stream -> writePropertiesMarkdownFile(new InstanceProperties(), instancePropertyGroup, stream));
+            } catch (Exception e) {
+                System.out.println("Unable to write property file for group: " + instancePropertyGroup.getName());
+            }
+        });
+
+        //---------- Table Properties ----------
+        Path tablePath = Files.createDirectories(headPath.resolve("table/"));
+        TablePropertyGroup.getAll().forEach(tablePropertyGroup -> {
+            try {
+                writeFile(tablePath.resolve(groupNameToFileName(tablePropertyGroup.getName())),
+                        stream -> writePropertiesMarkdownFile(new TableProperties(new InstanceProperties()), tablePropertyGroup, stream));
+            } catch (Exception e) {
+                System.out.println("Unable to write property file for group: " + tablePropertyGroup.getName());
```

Comment:
It looks important to fail the build if an exception happens here, because it means the file didn't write properly. It looks like the exception should be thrown all the way out?